api/csi_ll_pmp-spec.yaml

```diff
@@ -0,0 +1,127 @@
+module:
+  name: Low-Level PMP API
+  description: >
+    This module provides functionality for PMP configuration.
+  c-specific: true
+  c-filename: csi_ll_pmp.h
+  c-include-files:
+  - filename: csi_types.h
+    system-header: false
+
+  c-definitions:
+  - comment: >
+      Attribute definitions of PMP entry:
+      PMP_ATTR_FLAG_R: indicate that the PMP entry permits to read
+      PMP_ATTR_FLAG_W: indicate that the PMP entry permits to write
+      PMP_ATTR_FLAG_X: indicate that the PMP entry permits to execute
+      PMP_ATTR_FLAG_L: indicate that the PMP entry permits to lock
+    fragment: |
+      #define PMP_ATTR_FLAG_R				    0x01
+      #define PMP_ATTR_FLAG_W				    0x02
+      #define PMP_ATTR_FLAG_X				    0x04
+      #define PMP_ATTR_FLAG_L				    0x80
+  - comment: >
+      Address-matching definitions of PMP entry:
+      PMP_ATTR_ADDR_MATCHING_DISABLED: indicate that the PMP entry is disabled
+      PMP_ATTR_ADDR_MATCHING_TOR: indicate that the PMP entry's address-matching is TOR
+      PMP_ATTR_ADDR_MATCHING_NA4: indicate that the PMP entry's address-matching is NA4
+      PMP_ATTR_ADDR_MATCHING_NAPOT: indicate that the PMP entry's address-matching is
+      NAPOT
+    fragment: |
+      #define PMP_ATTR_ADDR_MATCHING_DISABLED 0x0
+      #define PMP_ATTR_ADDR_MATCHING_TOR		  0x08
+      #define PMP_ATTR_ADDR_MATCHING_NA4		  0x10
+      #define PMP_ATTR_ADDR_MATCHING_NAPOT	  0x18
+
+  functions:
+  - name: csi_pmp_get_num_entries
+    description: >
+      Get the number of PMP entries for the hard id.
+      If the hart id is invalid, such as out of support range, it should return
+      an error
+    c-params:
+    - name: hart_id
```

Comment:
Why `hart_id` is required, if you running on one hart, you can only access to the pmp csrs belong to you, so this hart_id, I think it is not useful.

Reply:
My intention in reserving this parameter is that perhaps users may use different method to get the number of pmp entries. Such as using constant define in a macro. And the pmp entries of each core may be different.